mozilla_django_oidc/auth.py

```diff
@@ -157,7 +157,7 @@ def retrieve_matching_jwk(self, token):
 
         key = None
         for jwk in jwks['keys']:
-            if jwk['kid'] != smart_text(header.kid):
+            if jwk['kid'] != smart_text(header.kid) and import_from_settings("OIDC_VERIFY_KID", True):
```

Comment:
This might be a better fix for #375 instead of #387 if #387 has security implications that are not acceptable.

For this to work, the parts need to be swapped though, because the `KeyError` will still happen otherwise:

```suggestion
            if import_from_settings("OIDC_VERIFY_KID", True) and jwk['kid'] != smart_text(header.kid):
```